CDCgov/prime-simplereport
# [Blocked] - waiting on async changes - patient bulk upload frontend

# FRONTEND PULL REQUEST

## Related Issue

- Allowing users to do bulk patient upload themselves 
- Resolves #4365 

## Changes Proposed

- Create a new page to handle patient bulk upload.
- New menu/dropdown button in manage patients page to get to the add individual patient & bulk upload pages.
- Update FontAwesome Icons.
- Remove the admin checkbox for patient bulk upload.

## Additional Information

- This cannot be merged until docs are complete for this change and has the appropriate changes made.
- FontAwesome was updated to get the new icons that needed to be used in this. This also means that a lot of the other icons have had slight design tweaks. In version 6, they also [no longer support internet explorer](https://fontawesome.com/v6/docs/web/setup/upgrade/whats-changed#at-a-glance). 
- Loader icon is fixed ‼️ (Resolves #4493 )

## Testing

- Run accessibility tests on new pages
- Ensure the new icons still look good
- Test upload with valid file and with invalid file 
[valid_file.csv](https://github.com/CDCgov/prime-simplereport/files/9888584/valid_file.csv)


## Screenshots / Demos

https://user-images.githubusercontent.com/10108172/198595250-7494a9c8-0b01-4a40-96dd-c6fd64af27f3.mp4
### Icon Changes (New on top, old below)
- Pagination (the right angle)
![image](https://user-images.githubusercontent.com/10108172/198675529-b365bce1-8dfd-4e5c-b453-e707198756cc.png)
![image](https://user-images.githubusercontent.com/10108172/198675663-7d1c8319-b999-44b1-ae06-a8fe796f1d10.png)
- Pagination (Left angle)
![image](https://user-images.githubusercontent.com/10108172/198675885-a8aee926-93f3-4e30-99d3-3a29cf18e8c0.png)
![image](https://user-images.githubusercontent.com/10108172/198675791-494c1b7d-e18e-46d3-9476-28376abe0530.png)

- Edit Patient (Plus icon)
![image](https://user-images.githubusercontent.com/10108172/198675311-c0c19679-65dd-4593-b2ac-f4cc337bb71e.png)
![image](https://user-images.githubusercontent.com/10108172/198675398-197d14bd-1902-4b80-a460-60ae2b3916d2.png)

- Test Queue (Close icon, Stopwatch (I can't see a difference), info circle)
![image](https://user-images.githubusercontent.com/10108172/198678093-f01e06c7-6115-4569-b0ea-6eaa11708639.png)
![image](https://user-images.githubusercontent.com/10108172/198678015-a9f784e8-6631-408f-9ab1-46395e02f6f8.png)

-Test Print Modal (Globe (The size is a tiny bit different))
![image](https://user-images.githubusercontent.com/10108172/198678538-a3489a30-8600-4f6d-adea-00aa9cf07ecb.png)
![image](https://user-images.githubusercontent.com/10108172/198678512-2af23977-ec09-4bbe-ac3a-d7baccb90474.png)
